toplevel/coqloop.ml

```diff
@@ -410,7 +415,8 @@ let rec vernac_loop ~state =
     let loc = Loc.get_loc info in
     let msg = CErrors.iprint (e, info) in
     TopErr.print_error_for_buffer ?loc Feedback.Error msg top_buffer;
-    vernac_loop ~state
+    if fast_death () then exit 1
+    else vernac_loop ~state
```

Comment:
I'd prefer:
```
if exit_on_error () then exit 1;
vernac_loop ~state
```
as to make the critical importance of tail-rec clear here.

Reply:
As you like.

Reply:
Thanks